services/sidecar/src/simcore_service_sidecar/utils.py

```diff
@@ -109,3 +109,21 @@ def touch_tmpfile(extension=".dat") -> Path:
     """
     with tempfile.NamedTemporaryFile(delete=False, suffix=extension) as file_handler:
         return Path(file_handler.name)
+
+
+def cancel_task(task_name: str) -> None:
+    tasks = asyncio.all_tasks()
+    logger.debug("running tasks: %s", tasks)
+    for task in tasks:
+        if task.get_name() == task_name:
+            logger.warning("canceling task %s....................", task)
+            task.cancel()
+
+
+def cancel_task_by_fct_name(fct_name: str) -> None:
+    tasks = asyncio.all_tasks()
+    logger.debug("running tasks: %s", tasks)
+    for task in tasks:
+        if task.get_coro().__name__ == fct_name:
+            logger.warning("canceling task %s....................", task)
+            task.cancel()
```

Comment:
Another optimisation proposal in case `task.get_coro().__name__` is unique
```suggestion
    tasks_function_names = {t.get_coro().__name__: t for t in tasks}
    if fct_name in tasks_function_names:
        task = tasks_function_names[fct_name]
        logger.warning("canceling task %s....................", task)
        task.cancel()
            task.cancel()
```

Reply:
same